tembo-cli/src/cli/tembo_config.rs

```diff
@@ -29,10 +30,148 @@ pub struct InstanceSettings {
         default = "default_extensions"
     )]
     pub extensions: Option<HashMap<String, Extension>>,
+    pub app_services: Option<HashMap<String, AppType>>,
     pub extra_domains_rw: Option<Vec<String>>,
     pub ip_allow_list: Option<Vec<String>>,
 }
 
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum AppType {
+    #[serde(rename = "restapi")]
+    RestAPI(Option<AppConfig>),
+    #[serde(rename = "http")]
+    HTTP(Option<AppConfig>),
+    #[serde(rename = "mq-api")]
+    MQ(Option<AppConfig>),
+    #[serde(rename = "embeddings")]
+    Embeddings(Option<AppConfig>),
+    #[serde(rename = "custom")]
+    Custom(AppService),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct AppConfig {
+    pub env: Option<Vec<EnvVar>>,
+    pub resources: Option<ResourceRequirements>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
```

Comment:
lots of these structs are already in the `controller` crate. Do they need to be re-defined?

Reply:
Nice! Lemme look at that.

Reply:
@ChuckHend Done!